diff --git a/faiss/python/CMakeLists.txt b/faiss/python/CMakeLists.txt
index b28723761..92ba1b757 100644
--- a/faiss/python/CMakeLists.txt
+++ b/faiss/python/CMakeLists.txt
@@ -214,41 +214,41 @@ find_package(OpenMP REQUIRED)
 
 target_link_libraries(swigfaiss PRIVATE
   faiss
-  Python::Module
+  Python::SABIModule
   Python::NumPy
   OpenMP::OpenMP_CXX
 )
 
 target_link_libraries(swigfaiss_avx2 PRIVATE
   faiss_avx2
-  Python::Module
+  Python::SABIModule
   Python::NumPy
   OpenMP::OpenMP_CXX
 )
 
 target_link_libraries(swigfaiss_avx512 PRIVATE
   faiss_avx512
-  Python::Module
+  Python::SABIModule
   Python::NumPy
   OpenMP::OpenMP_CXX
 )
 
 target_link_libraries(swigfaiss_avx512_spr PRIVATE
   faiss_avx512_spr
-  Python::Module
+  Python::SABIModule
   Python::NumPy
   OpenMP::OpenMP_CXX
 )
 
 target_link_libraries(swigfaiss_sve PRIVATE
   faiss_sve
-  Python::Module
+  Python::SABIModule
   Python::NumPy
   OpenMP::OpenMP_CXX
 )
 
 target_link_libraries(faiss_example_external_module PRIVATE
-  Python::Module
+  Python::SABIModule
   Python::NumPy
   OpenMP::OpenMP_CXX
   swigfaiss
@@ -265,7 +265,7 @@ target_include_directories(swigfaiss_sve PRIVATE ${PROJECT_SOURCE_DIR}/../..)
 target_include_directories(faiss_example_external_module PRIVATE ${PROJECT_SOURCE_DIR}/../..)
 
 find_package(Python REQUIRED
-  COMPONENTS Development NumPy
+  COMPONENTS Development.SABIModule NumPy
 )
 
 add_library(faiss_python_callbacks EXCLUDE_FROM_ALL
